src/openforms/forms/tests/variables/test_variables_in_logic_trigger.py

```diff
@@ -117,6 +116,57 @@ def test_invalid_user_defined_variable_in_trigger(self):
             error["invalidParams"][0]["reason"],
         )
 
+    def test_invalid_date_format_for_user_defined_variable(self):
+        user = SuperUserFactory.create()
+        form = FormFactory.create()
+        FormVariableFactory.create(
+            source=FormVariableSources.user_defined,
+            key="date_var",
+            form=form,
+            data_type=FormVariableDataTypes.date,
+        )
+        FormStepFactory.create(
+            form=form,
+            form_definition__configuration={
+                "components": [
+                    {
+                        "type": "textfield",
+                        "key": "textfield",
+                    },
+                ]
+            },
+        )
+        form_logic_data = [
+            {
+                "form": f"http://testserver{reverse('api:form-detail', kwargs={'uuid_or_slug': form.uuid})}",
+                "order": 0,
+                "json_logic_trigger": {"==": [{"var": "textfield"}, "some test value"]},
```

Comment:
Then, if you remove the textfield, here you can use some trigger `{"!!": [True]}` which always evaluates to true.